### PR TITLE
Light up for newer arm64 hardware capabilities

### LINKED ISSUE #89937: NativeAOT on Graviton2 As `armv8.2-a` Fails to Execute
### Description

When NativeAOT-compiling an application for deployment onto AWS Lambda, specifying `IlcInstructionSet` with the known, documented supported instruction sets for Graviton2 processors fails to execute with the error message:

> The required instruction sets are not supported by the current CPU.

### Reproduction Steps

For a minimal reproduction, I'm using the code from [maxday's lambda-perf cold start–testing repository](https://github.com/maxday/lambda-perf/tree/main/s3-uploader/runtimes/dotnet7_aot_on_provided_al2/src), but targeting .NET 8's NativeAOT. The meaningful parts of that are:

```csharp
public class Function
{
    private static async Task Main()
    {
        await LambdaBootstrapBuilder.Create(FunctionHandler, new SourceGeneratorLambdaJsonSerializer<LambdaFunctionJsonSerializerContext>())
            .Build()
            .RunAsync();
    }

    public static StatusResponse FunctionHandler()
    {
        return new StatusResponse(statusCode: 200);
    }
}

public record StatusResponse(int statusCode);

[JsonSerializable(typeof(StatusResponse))]
public partial class LambdaFunctionJsonSerializerContext : JsonSerializerContext
{
}
```

…and for .csproj settings:

```xml
    <TargetFramework>net8.0</TargetFramework>
    <RuntimeIdentifier>linux-arm64</RuntimeIdentifier>
    <PublishAot>true</PublishAot>
    <AssemblyName>bootstrap</AssemblyName>

    <IlcInstructionSet>armv8.2-a</IlcInstructionSet>
```

### Expected behavior

Because Amazon's Graviton2 processor is [documented](https://github.com/aws/aws-graviton-getting-started#building-for-graviton2-graviton3-and-graviton3e) as revision 8.2-a with support for features "fp16, rcpc, dotprod, crypto" and `ilc` doesn't support "fp16" or "crypto", I expected this:

```xml
<IlcInstructionSet>armv8.2-a,rcpc,dotprod</IlcInstructionSet>
```

…to compile and run. It compiled, but did not run. Figuring that maybe I was trying to add too much sauce, I tried:

```xml
<IlcInstructionSet>armv8.2-a</IlcInstructionSet>
```

…expecting this to compile and run. It compiled, but did not run.

### Actual behavior

Upon execution, this error message is received:

> The required instruction sets are not supported by the current CPU.

### Regression?

_No response_

### Known Workarounds

Leave off `IlcInstructionSet`, implicitly targeting revision 8.0. This does compile (natch) and run.

### Configuration

I am cross-compiling for arm64 in the Docker image `mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-2.0-cross-arm64`:

```bash
~/.dotnet/dotnet publish -p:SysRoot=/crossrootfs/arm64 -p:LinkerFlavor=lld
```

`dotnet --version` is "8.0.100-rc.1.23402.12"

### Other information

I admit that I'm not certain this is a problem with the NativeAOT compiler, since I don't know how to look into a binary and say "Aha, here is the incorrect instruction."

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #89937

### CODE REVIEW

File: src/native/minipal/cpufeatures.c [hunk 1]
```diff
@@ -17,8 +17,21 @@
 #include "minipalconfig.h"
 
 #if HAVE_AUXV_HWCAP_H
+
 #include <sys/auxv.h>
 #include <asm/hwcap.h>
+
+// Light-up for hardware capabilities that are not present in older headers used by the portable build.
+#ifndef HWCAP_ASIMDRDM
```
Comment (jkotas) [inline src/native/minipal/cpufeatures.c:25]: We do similar light-up in other places, e.g. here: https://github.com/dotnet/runtime/blob/9cf693a66061b7c1efeeed7053a321fed786d3ac/src/coreclr/pal/src/thread/process.cpp#L67

**Review by MichalStrehovsky — APPROVED** [review]
This was not introduced by the unification with NativeAOT - it was a pre-existing issue, right? I see the `HWCAP_LRCPC` ifdefs in old jitsupport.cpp as well (and they probably originate from there).

**Comment by jkotas** [comment]
Right, we had this bug in official builds since forever.

**Review by EgorBo — APPROVED** [review]

**Comment by EgorBo** [comment]
Looks like it improved a bunch of benchmarks on Linux-arm64 🙂 

- https://github.com/dotnet/perf-autofiling-issues/issues/20508

**Pull request merged**